### ISSUE TITLE
Crash when deleting repository

### ISSUE THREAD

**Issue by wadefelix** [goharbor/harbor#12339]
If you are reporting a problem, please make sure the following information are provided:

**Expected behavior and actual behavior:**
A clear and concise description of what you expected to happen and what's the actual behavior. If applicable, add screenshots to help explain your problem.

**Steps to reproduce the problem:**
Please provide the steps to reproduce this problem.

**Versions:**
    Image:          goharbor/harbor-core:v2.0.0
    Image ID:       docker-pullable://goharbor/harbor-core@sha256:75b5900d2335c87ca34ee80e64e2a6b56cbe218aceaef8614efc31594ad84d38

**Additional context:**

- **Harbor config files:** You can get them by packaging `harbor.yml` and files in the same directory, including subdirectory.
- **Log files:** You can get them by package the `/var/log/harbor/` .
```
2020-06-28T05:54:40Z [ERROR] [/controller/event/handler/webhook/artifact/artifact.go:117]: failed to get repository with name library/ayena: repository library/ayena not found
2020-06-28T05:54:40Z [INFO] [/pkg/notifier/notifier.go:204]: Handle notification with topic 'DELETE_ARTIFACT': &event.DeleteArtifactEvent{ArtifactEvent:(*event.ArtifactEvent)(0xc0003aa4d0)}
2020-06-28T05:54:40Z [INFO] [/pkg/notifier/notifier.go:204]: Handle notification with topic 'DELETE_ARTIFACT': &event.DeleteArtifactEvent{ArtifactEvent:(*event.ArtifactEvent)(0xc00055ce70)}
2020-06-28T05:54:40Z [ERROR] [/controller/event/handler/webhook/artifact/artifact.go:62]: failed to get project:1, error: pq: sorry, too many clients already
2020-06-28T05:54:40Z [ERROR] [/pkg/notifier/notifier.go:202]: Error occurred when triggering handler *artifact.Handler of topic DELETE_ARTIFACT: pq: sorry, too many clients already
2020-06-28T05:54:40Z [ERROR] [/controller/event/handler/webhook/artifact/artifact.go:117]: failed to get repository with name library/ayena: repository library/ayena not found
2020-06-28T05:54:40Z [INFO] [/pkg/notifier/notifier.go:204]: Handle notification with topic 'DELETE_ARTIFACT': &event.DeleteArtifactEvent{ArtifactEvent:(*event.ArtifactEvent)(0xc000406af0)}
2020-06-28T05:54:40Z [ERROR] [/controller/event/handler/webhook/artifact/artifact.go:62]: failed to get project:1, error: pq: sorry, too many clients already
2020-06-28T05:54:40Z [ERROR] [/pkg/notifier/notifier.go:202]: Error occurred when triggering handler *artifact.Handler of topic DELETE_ARTIFACT: pq: sorry, too many clients already
2020-06-28T05:54:40Z [ERROR] [/controller/event/handler/webhook/artifact/artifact.go:117]: failed to get repository with name library/ayena: repository library/ayena not found
2020-06-28T05:54:40Z [INFO] [/pkg/notifier/notifier.go:204]: Handle notification with topic 'DELETE_ARTIFACT': &event.DeleteArtifactEvent{ArtifactEvent:(*event.ArtifactEvent)(0xc001f3e0e0)}
2020-06-28T05:54:40Z [ERROR] [/pkg/notifier/notifier.go:202]: Error occurred when triggering handler *replication.Handler of topic DELETE_ARTIFACT: pq: sorry, too many clients already
2020-06-28T05:54:40Z [INFO] [/pkg/notifier/notifier.go:204]: Handle notification with topic 'DELETE_ARTIFACT': &event.DeleteArtifactEvent{ArtifactEvent:(*event.ArtifactEvent)(0xc000383f80)}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x98 pc=0xaaca26]

goroutine 2425934 [running]:
github.com/goharbor/harbor/src/common/models.(*Project).GetMetadata(...)
        /harbor/src/common/models/project.go:52
github.com/goharbor/harbor/src/common/models.(*Project).IsPublic(0x0, 0x0)
        /harbor/src/common/models/project.go:69 +0x26
github.com/goharbor/harbor/src/controller/event/handler/webhook/artifact.(*Handler).constructArtifactPayload(0xc0002ee278, 0xc0004123f0, 0x1d1845d, 0xf, 0xc000fec0d0)
        /harbor/src/controller/event/handler/webhook/artifact/artifact.go:94 +0x69
github.com/goharbor/harbor/src/controller/event/handler/webhook/artifact.(*Handler).handle(0xc0002ee278, 0xc0004123f0, 0xf1020e, 0x308daa0)
        /harbor/src/controller/event/handler/webhook/artifact/artifact.go:75 +0x210
github.com/goharbor/harbor/src/controller/event/handler/webhook/artifact.(*Handler).Handle(0xc0002ee278, 0x1a65fa0, 0xc00062c038, 0xab397b, 0xc00192ded8)
        /harbor/src/controller/event/handler/webhook/artifact/artifact.go:46 +0x107
github.com/goharbor/harbor/src/pkg/notifier.(*NotificationWatcher).Notify.func1.1(0x208fca0, 0xc0002ee278, 0x0, 0x1d1845d, 0xf, 0x1a65fa0, 0xc00062c038)
        /harbor/src/pkg/notifier/notifier.go:200 +0xba
created by github.com/goharbor/harbor/src/pkg/notifier.(*NotificationWatcher).Notify.func1
        /harbor/src/pkg/notifier/notifier.go:194 +0xac
```

**Comment by ninjadq** [comment]
The log shows `library/ayena` not exist. It seems that the image already deleted. Could you verify it?
It also shows that too many clients to connect database. Are there any other logs or context here? like logs in the database?

**Comment by mmpei** [comment]
fixed in release 2.0.2 and master.
